src/encoded/static/components/viz/AreaChart.js

```diff
@@ -566,10 +566,16 @@ export class HorizontalD3ScaleLegend extends React.Component {
         return false;
     }
 
+    componentDidUpdate(pastProps, pastState) {
+        setTimeout(function () {
+            ReactTooltip.rebuild();
+        }, 1000);
+    }
```

Comment:
Is there any way to configure this delay so that it's not hard-coded? Is there a particular part of the interface that is changing in response to a change in props or state that is resulting in the necessity to rebuild the tooltips? If so, using a conditional here instead of a timeout will probably be better in the long run.

Reply:
good catch! let me check it. thanks!

Reply:
@Bianca-Morris  I tested it once more. The `shouldComponentUpdate(nextProps, nextState)` life-cycle function prevents unnecessary `componentDidUpdate` calls, so the `ReactTooltip.rebuild` is called when only the color scale or terms change.